canvas/gradient.go

```diff
@@ -0,0 +1,190 @@
+package canvas
+
+import (
+	"image"
+	"image/color"
+	"image/draw"
+	"math"
+
+	"fyne.io/fyne"
+)
+
+// Gradient describes a gradient between two colors
+type Gradient struct {
+	baseObject
+
+	Generator func(w, h int) image.Image
+
+	Translucency float64 // Set a translucency value > 0.0 to fade the raster
+
+	img draw.Image // internal cache for pixel generator
+}
+
+// Alpha is a convenience function that returns the alpha value for a raster
+// based on its Translucency value. The result is 1.0 - Translucency.
+func (g *Gradient) Alpha() float64 {
```

Comment:
I also think we can do without this as alpha is set by the colours passed to the generator